wait/host.go

```diff
@@ -55,6 +55,27 @@ func (a *HostAwaitility) WaitForMasterUserRecord(name string, criteria ...Master
 	return mur, err
 }
 
+// WaitForNotification waits until there is the notification with the given name available
+func (a *HostAwaitility) WaitForNotification(name string) (*toolchainv1alpha1.Notification, error) {
+	notification := &toolchainv1alpha1.Notification{}
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		notification = &toolchainv1alpha1.Notification{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, notification); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of notification '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		if notification.Name != "" && notification.Spec.UserID != "" {
```

Comment:
Why do we need that check? notification.Name and Spec.UserID can't be nil here, right?

Reply:
@alexeykazakov sorry, this wasn't really ready for review, but i've changed the PR to open. The tests are failing for me which is why i wasn't going to open it yet lol. But i will address these comments soon

Reply:
fixed here: 46d9849d6db6dfe9991c5a66f4a3e53347166d56